src/metabase/legacy_mbql/schema.cljc

```diff
@@ -638,6 +638,10 @@
   x [:or NumericExpressionArg
      StringExpressionArg])
 
+(defclause ^{:requires-features #{:expressions :expressions/float}} float
+  x [:or NumericExpressionArg
```

Comment:
Why do we allow numeric expressions here?

Reply:
You're right! Good catch.